src/vs/workbench/contrib/terminalContrib/chat/browser/terminal.initialHint.contribution.ts

```diff
@@ -0,0 +1,297 @@
+import { IDetachedTerminalInstance, ITerminalContribution, ITerminalInstance, ITerminalService, IXtermTerminal } from 'vs/workbench/contrib/terminal/browser/terminal';
+import { registerTerminalContribution } from 'vs/workbench/contrib/terminal/browser/terminalExtensions';
+import type { Terminal as RawXtermTerminal, IDecoration, ITerminalAddon } from '@xterm/xterm';
+import { TerminalWidgetManager } from 'vs/workbench/contrib/terminal/browser/widgets/widgetManager';
+import { ITerminalProcessManager, ITerminalProcessInfo } from 'vs/workbench/contrib/terminal/common/terminal';
+import { ITerminalCapabilityStore, TerminalCapability } from 'vs/platform/terminal/common/capabilities/capabilities';
+import { IKeybindingService } from 'vs/platform/keybinding/common/keybinding';
+import { localize } from 'vs/nls';
+import { Emitter, Event } from 'vs/base/common/event';
+import { OS } from 'vs/base/common/platform';
+import { KeybindingLabel } from 'vs/base/browser/ui/keybindingLabel/keybindingLabel';
+import { IContentActionHandler, renderFormattedText } from 'vs/base/browser/formattedTextRenderer';
+import { WorkbenchActionExecutedClassification, WorkbenchActionExecutedEvent } from 'vs/base/common/actions';
+import { AccessibilityVerbositySettingId } from 'vs/workbench/contrib/accessibility/browser/accessibilityConfiguration';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { ICommandService } from 'vs/platform/commands/common/commands';
+import { IProductService } from 'vs/platform/product/common/productService';
+import { ITelemetryService } from 'vs/platform/telemetry/common/telemetry';
+import { IInlineChatService, IInlineChatSessionProvider, InlineChatProviderChangeEvent } from 'vs/workbench/contrib/inlineChat/common/inlineChat';
+import { status } from 'vs/base/browser/ui/aria/aria';
+import * as dom from 'vs/base/browser/dom';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { TerminalChatCommandId } from 'vs/workbench/contrib/terminalContrib/chat/browser/terminalChat';
+import { TerminalInstance } from 'vs/workbench/contrib/terminal/browser/terminalInstance';
+import { TerminalSettingId } from 'vs/platform/terminal/common/terminal';
+import 'vs/css!./media/terminalInitialHint';
+
+const $ = dom.$;
+
+export class InitialHintAddon extends Disposable implements ITerminalAddon {
+	private readonly _onDidRequestCreateHint = this._register(new Emitter<void>());
+	get onDidRequestCreateHint(): Event<void> { return this._onDidRequestCreateHint.event; }
+	private readonly _disposables = this._register(new MutableDisposable<DisposableStore>());
+
+	constructor(private readonly _capabilities: ITerminalCapabilityStore,
+		private readonly _onDidChangeProviders: Event<InlineChatProviderChangeEvent>) {
+		super();
+		this._disposables.value = this._register(new DisposableStore());
```

Comment:
Can this move into activate? Only need to create the DisposableStore once activate happens. You can also avoid the `?` by doing:

```ts
const store = new DisposableStore

...store.add(...)

this._disposables.value = store
```